pallets/credit/src/lib.rs

```diff
@@ -696,23 +738,22 @@ pub mod pallet {
                 score_delta
             );
             if score_delta > 0 {
-                let onboard_era = Self::get_onboard_era(&server_id);
-                if onboard_era.is_none() {
-                    // credit is not updated if the device is never online
-                    return;
-                }
                 let current_era = Self::get_current_era();
-                // if this is the first update, we use onboard era as the last update era
-                let last_credit_update_era =
-                    Self::last_credit_update(&server_id).unwrap_or(onboard_era.unwrap());
-                let mut eras = (current_era - last_credit_update_era) as u64;
-                if eras < 2 && Self::last_credit_update(&server_id).is_none() {
+                let now_as_secs = T::UnixTime::now().as_secs();
+                let (mut time_eras, era_used) = Self::check_update_credit_interval(
+                    &server_id,
+                    current_era,
+                    onboard_era.unwrap(),
+                    now_as_secs,
+                );
+
+                if time_eras < 2 && Self::last_credit_update_timestamp(&server_id).is_none() {
```

Comment:
looks duplicated, can we put this check into `check_update_credit_interval`?

Reply:
it is for different usage. for new comming, giving them relexed update contition